pkg/util/traceid.go

```diff
@@ -66,3 +66,14 @@ func EqualHexStringTraceIDs(a, b string) (bool, error) {
 
 	return bytes.Equal(aa, bb), nil
 }
+
+func PadTraceIDTo16Bytes(traceID []byte) []byte {
+	if len(traceID) == 16 {
+		return traceID
+	}
+
+	padded := make([]byte, 16-len(traceID))
```

Comment:
likely faster to make a slice of 16 bytes and copy the traceID into it.

Reply:
oh, and i think this will panic if len > 16. we should probably just take the least significant 16 bytes if something > 16 shows up here.